src/Components/FacetBreadcrumbs.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import Button from "@material-ui/core/IconButton";
+import {
+  BreadcrumbManager as BreadcrumbManagerType,
+  BreadcrumbManagerState,
+  buildBreadcrumbManager
+} from "@coveo/headless";
+import { headlessEngine } from "../Engine";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import Link from "@material-ui/core/Link";
+import ClearIcon from "@material-ui/icons/Clear";
+
+export default class FacetBreadcrumbs extends React.Component {
+  private headlessBreadcrumbManager: BreadcrumbManagerType;
+  state: BreadcrumbManagerState;
+
+  constructor(props: any) {
+    super(props);
+
+    this.headlessBreadcrumbManager = buildBreadcrumbManager(headlessEngine);
+
+    this.state = this.headlessBreadcrumbManager.state;
+  }
+  componentDidMount() {
+    this.headlessBreadcrumbManager.subscribe(() => this.updateState());
+  }
+
+  updateState() {
+    this.setState(this.headlessBreadcrumbManager.state);
+  }
+
+  getSelectedValues(key: string, objId: string) {
+    const selectedValuesObj: object = this.state[key][objId]["values"];
+    const selectedValues: Array<string> = [];
+    for (let selectedKey in selectedValuesObj) {
+      selectedValues.push(selectedValuesObj[selectedKey]);
+    }
+
+    return selectedValues;
+  }
+
+  render() {
+    let breadcrumbs: object = {};
```

Comment:
I personally prefer to keep as much logic as possible **outside** the render function. Because of this, I think we should put this chunk of logic in its own clearly named method (perhaps something like `getBreadcrumbObjects`?). What do you think? 🙂